### PR TITLE
Bomb spell

### PR DESCRIPTION
**Pull request by OwenBoisse**
bomb spell
- currently hit scan
- explodes after 5 seconds dealing damage to boss if in range

### CODE REVIEW

File: src/spells.rs [hunk 1]
```diff
@@ -111,6 +152,132 @@ fn handle_lightning(
...
+                bomb_hit_effect,
+                &explosion_entity,
+                boss_health.get_mut(boss_entity),
+                players.get(boss_entity),
+                &mut next_phase,
+                &current_phase,
+            );
+        }
+        if bomb_timer.timer.tick(time.delta()).finished() {
+            commands.entity(explosion_entity).despawn();
```
[the remark this answers is not in the document]
Comment (OwenBoisse) [reply src/spells.rs:276]: uh kinda, if you delete it straight away it just wont detect any collisions

**Review by rosefromthedead — CHANGES_REQUESTED** [review]
I'd like to block this until #48 is landed

File: src/spells.rs [hunk 2]
```diff
@@ -111,6 +152,132 @@ fn handle_lightning(
...
+                    CollisionLayers::new(
+                        PhysLayer::PlayerProjectile,
+                        (((LayerMask::ALL ^ PhysLayer::Player) ^ PhysLayer::BossProjectile)
+                            ^ PhysLayer::Terrain)
+                            ^ PhysLayer::PlayerProjectile,
+                    ),
+                    BombTimer {
+                        timer: Timer::new(Duration::from_secs(1), TimerMode::Once),
+                    },
+                    ProjectileHitEffect::Damage(DamageMask::FIRE, 25.0),
```
[the remark this answers is not in the document]
Comment (AmosHolland) [reply src/spells.rs:325]: None of the changes to projectiles really affect this. Either change not to use a projectile, or make the change that amin suggested, where you make a projectile effect that doesn't despawn the projectile. Probably the latter.

**Review by AmosHolland — APPROVED** [review]
This looks good, nice work!

**Review by AmosHolland — APPROVED** [review]
LGTM!

**Review by rosefromthedead — APPROVED** [review]

**Pull request merged**